app/src/docs/_examples/tables/PagedTable.example.tsx

```diff
@@ -39,37 +47,46 @@ export default function PagedTable() {
         [],
     );
 
-    const api = useCallback(
+    const api: LazyDataSourceApi<Person, number, unknown> = useCallback(
         async (rq: LazyDataSourceApiRequest<{}>) => {
             const result = await svc.api.demo.personsPaged({
                 ...rq,
                 filter: { departmentId: 13 }, // to get less results and non round-numbered number of people
-                page: state.page - 1, // server counts from 0, UI - from 1
-                pageSize: state.pageSize,
             });
-            setState((s) => ({ ...s, totalCount: result.totalCount }));
-            result.count = result.items.length;
-            result.from = 0;
             return result;
         },
-        [state.page, state.pageSize],
+        [svc.api.demo],
     );
 
     const dataSource = useLazyDataSource<Person, number, unknown>({
         api,
+        rowOptions: {
+            checkbox: {
+                isVisible: true,
+            },
+        },
         backgroundReload: true,
-    }, [state.page]);
-    const view = dataSource.useView(state, setState, {});
+    }, []);
+
+    const view = dataSource.useView(state, setTableState, {});
+    const listProps = view.getListProps();
 
     return (
         <Panel background="surface" shadow cx={ css.container }>
-            <DataTable { ...view.getListProps() } getRows={ view.getVisibleRows } value={ state } onValueChange={ setState } columns={ columns } headerTextCase="upper" />
+            <DataTable
+                { ...listProps }
+                getRows={ view.getVisibleRows }
+                value={ state }
+                onValueChange={ setTableState }
+                columns={ columns }
+                headerTextCase="upper"
+            />
             <FlexRow size="36" padding="12">
                 <FlexSpacer />
                 <Paginator
                     value={ state.page }
-                    onValueChange={ (newPage) => setState({ ...state, page: newPage, scrollTo: { index: 0 } }) }
-                    totalPages={ Math.ceil(state.totalCount / state.pageSize) }
+                    onValueChange={ (newPage) => setTableState({ ...state, page: newPage, scrollTo: { index: 0 } }) }
```

Comment:
Setting scrollTo can now be moved to the new setTableState function, to have all state-setting logic at one place.

Reply:
`setTableState` is used in other places too, so, it can cause overriding of a new state in case of `pageSize` ~ 100 (like, you won't be able to scroll to 21'st item, because scrollTo has a new link and VirtualList will be forced to scroll to that position.

https://github.com/epam/UUI/blob/6fa0ee3a88f345ce8433d730f8e46147d567bc69/app/src/docs/_examples/tables/PagedTable.example.tsx#L80
https://github.com/epam/UUI/blob/6fa0ee3a88f345ce8433d730f8e46147d567bc69/app/src/docs/_examples/tables/PagedTable.example.tsx#L71


@jakobz, WDYT?

Reply:
Got it. Let's keep as is.